ambari-server/src/main/java/org/apache/ambari/server/orm/entities/MpackInstanceEntity.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ambari.server.orm.entities;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import javax.annotation.Nullable;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.DiscriminatorColumn;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Inheritance;
+import javax.persistence.InheritanceType;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import javax.persistence.TableGenerator;
+
+/**
+ * Entity to encapsulate a blueprint's or cluster template's use of an mpack. It contains the mpack name, version, url
+ * referencing a management pack from the blueprint. The reference contains the name and
+ * the version of the mpack, but no direct database reference to the mpack entity as a blueprint
+ * can be saved without the referenced mpack being present.
+ */
+@Entity
+@Table(name = "mpack_instance")
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
+@DiscriminatorColumn(name = "owner", length = 20)
+@TableGenerator(name = "mpack_instance_id_generator", table = "ambari_sequences", pkColumnName = "sequence_name",
+  valueColumnName = "sequence_value", pkColumnValue = "mpack_instance_id_seq", initialValue = 1)
+public abstract class MpackInstanceEntity {
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.TABLE, generator = "mpack_instance_id_generator")
+  @Column(name = "id", nullable = false, updatable = false)
+  private Long id;
+
+  @Column(name = "mpack_name", nullable = false)
+  private String mpackName;
+
+  @Column(name = "mpack_version", nullable = false)
+  private String mpackVersion;
+
+  @Column(name = "mpack_uri")
+  private String mpackUri;
+
+  @OneToMany(cascade = CascadeType.ALL, mappedBy = "mpackInstance")
+  private Collection<MpackInstanceServiceEntity> serviceInstances = new ArrayList<>();
```

Comment:
Why make these a Collection instead of a List?

Reply:
I prefer using the most general suitable interface.